src/lib/deck.js

```diff
@@ -1,30 +1,32 @@
 class Deck {
     constructor(cards) {
         this._cards = cards; // array of Cards
-        this._length = this._cards.length // does this get updated when draw changes the # of cards?)
     }
 
     draw(quantity=1) {
         const drawn = this._cards.slice(0, quantity);
-        this._cards = this._cards.slice(quantity, this._length);
+        this._cards = this._cards.slice(quantity, this._cards.length);
         return drawn;
     }
 
     shuffle(times=1) {
-        // swap two random cards as many times as there are cards * times
-        for (let i = 0; i < this._length * times; i++) {
-            const index1 = Math.floor((Math.random() * this._length));
-            const index2 = Math.floor((Math.random() * this._length));
-            const temp = this._cards[index1];
-
-            this._cards[index1] = this._cards[index2];
-            this._cards[index2] = temp;
-        }
-    }
+        let currIndex = this._cards.length * times;
+	    let temp, randIndex;
+
+	    while (currIndex !== 0) {
+
+		    randIndex = Math.floor(Math.random() * currIndex);
+		    currIndex -= 1;
+
+		    temp = this._cards[currIndex];
+		    this._cards[currIndex] = this._cards[randIndex];
+		    this._cards[randIndex] = temp;
```

Comment:
Destructuring swap example: 

```js
[this._cards[currIndex], this._cards[randIndex]] = [this._cards[randIndex], this._cards[currIndex]]
```

I'll leave it to you to determine if this is better.